src/bun.js/api/bun/socket.zig

```diff
@@ -1443,6 +1458,65 @@ fn NewSocket(comptime ssl: bool) type {
             globalObject.throw("Cannot construct Socket", .{});
             return null;
         }
+        pub fn resumeFromJS(this: *This, _: *JSC.JSGlobalObject, _: *JSC.CallFrame) JSValue {
+            JSC.markBinding(@src());
+
+            log("resume", .{});
+            if (this.flags.is_paused) {
+                this.flags.is_paused = !this.socket.resumeStream();
+            }
+            return .undefined;
+        }
+        pub fn pauseFromJS(this: *This, _: *JSC.JSGlobalObject, _: *JSC.CallFrame) JSValue {
+            JSC.markBinding(@src());
+
+            log("pause", .{});
+            if (!this.flags.is_paused) {
+                this.flags.is_paused = this.socket.pauseStream();
+            }
+            return .undefined;
+        }
+
+        pub fn setKeepAlive(this: *This, globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) JSValue {
+            JSC.markBinding(@src());
+            const args = callframe.arguments(2);
+
+            const enabled: bool = brk: {
+                if (args.len >= 1) {
+                    break :brk args.ptr[0].coerce(bool, globalThis);
+                }
+                break :brk false;
+            };
+
+            const initialDelay: u32 = brk: {
+                if (args.len > 1) {
+                    const signedDelay = args.ptr[1].coerce(i32, globalThis);
```

Comment:
use `validateIntegerRange` which will throw the correct error message and handle more edge cases, this can cause us to throw twice and then crash.

Reply:
fixed